trpcultivate_phenotypes/src/TripalCultivateValidator/ValidatorTraits/DataFileDelimiter.php

```diff
@@ -0,0 +1,52 @@
+<?php
+
+namespace Drupal\trpcultivate_phenotypes\TripalCultivateValidator\ValidatorTraits;
+
+/**
+ * Provides setters focused for setting a delimiter used by the importer
+ * to delimit data/values in the data file and getter to retrieve the set value.
+ */
+trait DataFileDelimiter {
+  // The key used to reference the delimiter set or get from the
+  // context property in the parent class.
+  private string $trait_key = 'delimiter';
+  
+  /**
+   * Sets the delimiter.
+   *
+   * @param string $delimiter
+   *   A value used to separate/delimit the values in a data file row or line. 
+   * 
+   * @throws \Exception
+   *   Invalid delimiter provided such as empty string, false and 0 values.
+   * 
+   * @return void
+   */
+  public function setDelimiter(string $delimiter) {
+    // Delimiter must have a value.
```

Comment:
I'm worried that if $delimiter is truly an empty string, then the current exception message won't reveal the actual problem. Instead, I would recommend a separate check for an empty string with an explicit exception message. Make sure to also update the test for this case!
```suggestion
    // Delimiter must not be an empty string
    if($delimiter === '') {
      throw new \Exception('The DataFileDelimiter Trait requires a non-empty string as a data file delimiter.');
    }
  
    // Delimiter must not be false, 0 or null
```